src/AnimationManager.ts

```diff
@@ -141,20 +141,39 @@ export class AnimationManager {
 			const info: AnimationParam = this._animateParam;
 			const self = this;
 			let prevPos = info.depaPos;
+			const directions = map(prevPos, (value, key) => {
+				return value <= info.destPos[key] ? 1 : -1;
+			});
+			let prevTime = new Date().getTime();
+			info.startTime = prevTime;
 
-			info.startTime = new Date().getTime();
 			(function loop() {
 				self._raf = null;
-				const easingPer = self.easing((new Date().getTime() - info.startTime) / param.duration);
-				const toPos: Axis = map(info.depaPos, (pos, key) => pos + info.delta[key] * easingPer);
-				const isCanceled = !self.em.triggerChange(toPos, false, prevPos);
+				const currentTime = new Date().getTime();
+				const prevEasingPer = self.easing((prevTime - info.startTime) / param.duration);
```

Comment:
@daybrush I think prevEasingPer could be processed like `prevTime` as follow. It would reduce the cost.

```js
prevPos = toPos;
prevTime = currentTime;
prevEasingPer = easingPer;
```